core/src/main/scala/com/microsoft/azure/synapse/ml/logging/common/SparkHadoopUtils.scala

```diff
@@ -0,0 +1,20 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.azure.synapse.ml.logging.common
+
+import org.apache.spark.SparkContext
+
+import com.microsoft.azure.synapse.ml.logging.SynapseMLLogging
+
+object SparkHadoopUtils {
+
+  def getHadoopConfig(key: String, sc: SparkContext): String = {
+      val value = sc.hadoopConfiguration.get(key, "")
+      if (value.isEmpty) {
+        SynapseMLLogging.logMessage(s"UsageUtils.getHadoopConfig: Hadoop configuration $key is empty.")
+        throw new IllegalArgumentException(s"UsageUtils.getHadoopConfig: Hadoop configuration $key is empty.")
+      }
+      value
+  }
```

Comment:
wont this just throw an error anyway if you dont include the second argument "" in .get

Reply:
Caution is taken to make sure they are not null. But, you are right, it's better to place a defensive peace in place. I am replacing argument with Option and also made sure spark context is created before calling get configuration on them.